lightly/api/api_workflow_upload_embeddings.py

```diff
@@ -70,21 +71,29 @@ def upload_embeddings(self, path_to_embeddings_csv: str, name: str):
         """
         # get the names of the current embeddings on the server:
         embeddings_on_server: List[DatasetEmbeddingData] = \
-            self.embeddings_api.get_embeddings_by_dataset_id(dataset_id=self.dataset_id)
-        names_embeddings_on_server = [embedding.name for embedding in embeddings_on_server]
-
-        if name in names_embeddings_on_server:
-            print(f"Aborting upload, embedding with name='{name}' already exists.")
-            self.embedding_id = next(embedding for embedding in embeddings_on_server if embedding.name == name).id
-            return
+            self._embeddings_api.get_embeddings_by_dataset_id(
+                dataset_id=self.dataset_id
+            )
+        embeddings_on_server_dict = {
+            embedding.name: embedding
+            for embedding in embeddings_on_server
+        }
+
+        if name in embeddings_on_server_dict.keys():
```

Comment:
This check probably won't work correctly. Did you try and upsize a dataset with this?

Because the name of the embedding in the API will be something like `default_DATE_TIME` whereas the name you are passing is only `default`

Reply:
I reworked it completely to use the startswith